interfaces/cython/cantera/onedim.py

```diff
@@ -387,20 +387,28 @@ def write_csv(self, filename, species='X', quiet=True):
         :param species:
             Attribute to use obtaining species profiles, e.g. ``X`` for
             mole fractions or ``Y`` for mass fractions.
+        :param normalize:
+            Boolean flag to indicate whether the mole/mass fractions should
+            be normalized (default is ``False``)
         """
 
         # save data
         cols = ('extra', 'T', 'D', species)
-        self.to_solution_array().write_csv(filename, cols=cols)
+        self.to_solution_array(normalize=normalize).write_csv(filename, cols=cols)
 
         if not quiet:
             print("Solution saved to '{0}'.".format(filename))
 
-    def to_solution_array(self, domain=None):
+    def to_solution_array(self, domain=None, normalize=False):
```

Comment:
While I agree that having the option to get the unnormalized composition is useful in certain circumstances, I'm really not convinced that it should be the default for extracting information from the 1D flame solver. Any deviations from a sum of 1 are just numerical noise, not anything deliberately set by the user, and the most notable feature will  be negative mass fractions below the absolute tolerance of the solver, which I don't think are that interesting to most users.